File: Models/Product.cs
namespace LearningPlayground.Models
{
    public class Product
    {
        public int ProductID { get; set; }
        public string Name { get; set; } = string.Empty;
        public string ProductNumber { get; set; } = string.Empty;
        public string? Color { get; set; }
        public decimal StandardCost { get; set; }
        public decimal ListPrice { get; set; }
        public string? Size { get; set; }
        public decimal? Weight { get; set; }
    }
}

File: ClientApp/src/app/demo/api/product.ts
interface InventoryStatus {
    label: string;
    value: string;
}
export interface Product {
    id?: string;
    code?: string;
    name?: string;
    description?: string;
    price?: number;
    quantity?: number;
    inventoryStatus?: InventoryStatus;
    category?: string;
    image?: string;
    rating?: number;
}
// 測試API用
export interface CustomProduct {
    productID: number,
    name: string,
    productNumber: string,
    color?: string,
    standardCost: number,
    listPrice: number,
    size?: string,
    weight?: number
}